File: src/main/java/br/com/correios/enderecador/telas/IncorporaDestinatarioTableModel.java
package br.com.correios.enderecador.telas;

import br.com.correios.enderecador.bean.DestinatarioBean;

import java.util.ArrayList;
import javax.swing.table.DefaultTableModel;

public class IncorporaDestinatarioTableModel extends DefaultTableModel {
    private final String[] colunas = new String[]{
        "Tratamento", "Empresa/Nome (linha 1)", "Empresa/Nome (linha 2)", "Caixa Postal", "Endereço", "Numero/Lote",
        "Complemento", "Bairro", "Cidade", "UF", "Email", "Telefone", "Fax", "CEP Caixa Postal", "Cep"};

    private ArrayList<DestinatarioBean> destinatarioBean = new ArrayList<>();

    public int getColumnCount() {
        return this.colunas.length;
    }

    public String getColumnName(int i) {
        return this.colunas[i];
    }

    public int getRowCount() {
        int retorno = 0;
        if (this.destinatarioBean != null)
            retorno = this.destinatarioBean.size();
        return retorno;
    }

    public Object getValueAt(int linha, int coluna) {
        DestinatarioBean destinatarioBean = this.destinatarioBean.get(linha);
        String valor = null;
        switch (coluna) {
            case 0:
                valor = destinatarioBean.getTitulo();
                break;
            case 1:
                valor = destinatarioBean.getNome();
                break;
            case 2:
                valor = destinatarioBean.getApelido();
                break;
            case 3:
                valor = destinatarioBean.getCaixaPostal();
                break;
            case 4:
                valor = destinatarioBean.getEndereco();
                break;
            case 5:
                valor = destinatarioBean.getNumeroEndereco();
                break;
            case 6:
                valor = destinatarioBean.getComplemento();
                break;
            case 7:
                valor = destinatarioBean.getBairro();
                break;
            case 8:
                valor = destinatarioBean.getCidade();
                break;
            case 9:
                valor = destinatarioBean.getUf();
                break;
            case 10:
                valor = destinatarioBean.getEmail();
                break;
            case 11:
                valor = destinatarioBean.getTelefone();
                break;
            case 12:
                valor = destinatarioBean.getFax();
                break;
            case 13:
                valor = destinatarioBean.getCepCaixaPostal();
                break;
            case 14:
                valor = destinatarioBean.getCep();
                break;
            default:
                valor = "";
                break;
        }
        if (valor == null)
            valor = "";
        return valor;
    }

    public boolean isCellEditable(int linha, int coluna) {
        return false;
    }

    public void setDestinatario(ArrayList<DestinatarioBean> destinatarioBean) {
        this.destinatarioBean = destinatarioBean;
        if (this.destinatarioBean == null)
            this.destinatarioBean = new ArrayList<>();
        fireTableDataChanged();
    }

    public DestinatarioBean getDestinatario(int linha) {
        DestinatarioBean destinatario = null;
        if (this.destinatarioBean.get(linha) != null)
            destinatario = this.destinatarioBean.get(linha);
        return destinatario;
    }

    public void setDestinatario(int linha, DestinatarioBean destinatarioBean) {
        this.destinatarioBean.set(linha, destinatarioBean);
        fireTableDataChanged();
    }
}

File: src/main/kotlin/br/com/correios/enderecador/telas/IncorporaDestinatarioTableModel.kt
package br.com.correios.enderecador.telas

import javax.swing.table.DefaultTableModel
import br.com.correios.enderecador.bean.DestinatarioBean
import java.util.ArrayList

class IncorporaDestinatarioTableModel : DefaultTableModel() {
    private val colunas = arrayOf(
        "Tratamento",
        "Empresa/Nome (linha 1)",
        "Empresa/Nome (linha 2)",
        "Caixa Postal", "Endereço",
        "Numero/Lote",
        "Complemento",
        "Bairro",
        "Cidade",
        "UF",
        "Email",
        "Telefone",
        "Fax",
        "CEP Caixa Postal",
        "Cep"
    )
    private var destinatarioBean: ArrayList<DestinatarioBean?>? = ArrayList()

    override fun getColumnCount(): Int {
        return colunas.size
    }

    override fun getColumnName(i: Int): String {
        return colunas[i]
    }

    override fun getRowCount(): Int {
        var retorno = 0
        if (destinatarioBean != null) retorno = destinatarioBean!!.size
        return retorno
    }

    override fun getValueAt(linha: Int, coluna: Int): String {
        val destinatarioBean = destinatarioBean!![linha]
        return when (coluna) {
            0 -> destinatarioBean!!.titulo
            1 -> destinatarioBean!!.nome
            2 -> destinatarioBean!!.apelido
            3 -> destinatarioBean!!.caixaPostal
            4 -> destinatarioBean!!.endereco
            5 -> destinatarioBean!!.numeroEndereco
            6 -> destinatarioBean!!.complemento
            7 -> destinatarioBean!!.bairro
            8 -> destinatarioBean!!.cidade
            9 -> destinatarioBean!!.uf
            10 -> destinatarioBean!!.email
            11 -> destinatarioBean!!.telefone
            12 -> destinatarioBean!!.fax
            13 -> destinatarioBean!!.cepCaixaPostal
            14 -> destinatarioBean!!.cep.orEmpty()
            else -> ""
        }
    }

    override fun isCellEditable(linha: Int, coluna: Int): Boolean {
        return false
    }

    fun setDestinatario(destinatarioBean: ArrayList<DestinatarioBean?>?) {
        this.destinatarioBean = destinatarioBean
        if (this.destinatarioBean == null) this.destinatarioBean = ArrayList()
        fireTableDataChanged()
    }

    fun getDestinatario(linha: Int): DestinatarioBean? {
        var destinatario: DestinatarioBean? = null
        if (destinatarioBean!![linha] != null) destinatario = destinatarioBean!![linha]
        return destinatario
    }

    fun setDestinatario(linha: Int, destinatarioBean: DestinatarioBean?) {
        this.destinatarioBean!![linha] = destinatarioBean
        fireTableDataChanged()
    }
}
